src/js/components/pipelines/ExpressionTextbox.jsx

```diff
@@ -0,0 +1,43 @@
+'use strict';
+var React = require('react'),
+    connect = require('react-redux').connect,
+    filterDataset = require('../../actions/datasetActions').filterDataset,
+    Property = require('../inspectors/Property');
+
+function mapStateToProps(state, ownProps) {
+  return {};
+}
+
+function mapDispatchToProps(dispatch, ownProps) {
+  return {
+    filterDataset: function(dsId, expression) {
+      dispatch(filterDataset(dsId, expression));
+    }
+  };
+}
+
+var ExpressionTextbox = React.createClass({
+  propTypes: {
+    dsId:  React.PropTypes.number
+  },
+
+  filter: function(evt) {
+    console.log("works");
+    this.props.filterDataset(this.props.dsId, evt);
+  },
+
+  render: function() {
+    var props = this.props,
+        id = props.dsId;
+
+    return (
+      <Property name="expressionProp" type="autocomplete"   autoType="expr" dsId={id} {...props} onChange={this.filter}>
+      </Property>
+    );
+  }
+});
+
+module.exports = {
+  connected: connect(mapStateToProps, mapDispatchToProps)(ExpressionTextbox),
+  disconnected: ExpressionTextbox
+}
```

Comment:
As we add more transforms, it's likely that each of them will have their own special inspector interface. So let's put these files under `src/js/components/pipelines/transforms` (and rename this particular file to `Filter.jsx` as being specific to the Filter UI?).